examples/acados_matlab_octave/linear_mpc/main.m

```diff
@@ -0,0 +1,163 @@
+%
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+% ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+% LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+% CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+% SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+% INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+% CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+% ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+% POSSIBILITY OF SUCH DAMAGE.;
+
+clear all; clc; close all;
+check_acados_requirements()
+
+%% solver settings
+h = 1/14;               % [s] sampling time
+N_horizon = 10;         % [-] number of prediction steps
+tf = N_horizon * h;     % [s] prediction horizon length
+
+[model,Ad,Bd] = quadcopter_model();     % system dynamics
+nx = length(model.x);                   % state size
+nu = length(model.u);                   % input size
+
+% ocp model
+ocp = AcadosOcp();
+ocp.name = 'quad_mpc';
+ocp.model = model;
+
+%% linear least squares cost
+% cost matrices
+Q = diag([0 0 10 10 10 10 0 0 0 5 5 5]);    % state cost
+R = 0.1*eye(nu);                            % input cost
+
+% details on discrete cost scaling:
+% https://docs.acados.org/python_interface/index.html#acados_template.acados_ocp_cost.AcadosOcpCost
+
+% initial cost (inputs only)
+ocp.cost.cost_type_0 = 'LINEAR_LS';
+ny_0 = nu;
+ocp.cost.Vu_0 = eye(ny_0);
+ocp.cost.Vx_0 = zeros(ny_0,nx);
+ocp.cost.W_0 = 1/h * 2 * R;  % scale to match the original cost
+ocp.cost.yref_0 = zeros(ny_0,1);
+
+% path cost (inputs and states)
+ocp.cost.cost_type = 'LINEAR_LS';
+ny = nu + nx;
+ocp.cost.Vu = [eye(nu); zeros(nx,nu)];
+ocp.cost.Vx = [zeros(nu, nx); eye(nx)];
+ocp.cost.W = 1/h * 2 * blkdiag(R,Q);  % scale to match the original cost
+ocp.cost.yref = zeros(ny, 1);
+
+% terminal cost (states only)
+ocp.cost.cost_type_e = 'LINEAR_LS';
+ny_e = nx;
+ocp.cost.Vx_e = eye(ny_e, nx);
+ocp.cost.W_e = 2 * Q;  % terminal cost is not scaled with h later
+ocp.cost.yref_e = zeros(ny_e, 1);
+
+%% constraints
+% input constraints
+u0 = 10.5916;                                       % steady-state input
+ocp.constraints.idxbu = 0:3;                        % zero-based indices
+ocp.constraints.lbu = [9.6; 9.6; 9.6; 9.6] - u0;    % input lower bounds
+ocp.constraints.ubu = [13; 13; 13; 13] - u0;        % input upper bounds
+
+% placeholder for the initial state constraint
+ocp.constraints.x0 = zeros(nx,1);
+
+% state constraints on the first, second and sixth state
+% (not applied to the initial state)
+ocp.constraints.idxbx = [0 1 5];                % zero-based indices
+infty = 1e6;                                    % to approximate one-sided constraints
```

Comment:
here we should use `get_acados_infty`, compare https://github.com/acados/acados/pull/1174

Reply:
right, thanks!